File: SOLUTIONS/LEETCODE/Vowels and Words/canBeTypedWords/canBeTypedWords.py
def canBeTypedWords(text: str, broken_letters: str) -> int:
  # Step 1: Convert brokenLetters into a set for quick lookup.
  # Why set? Because checking "is this letter broken?" is faster with a set.
  broken_set = set(broken_letters) 
  
  # Step 2: Split the text into words (words are separated by spaces).
  words = text.split()
  
  # Step 3: Initialize a counter for valid words.
  count = 0 
  
  # Step 4: Check each word one by one.
  for word in words:
    # Does this word contain any broken letter?
    if any(char in broken_set for char in word): 
      # if so, skip this word
      continue
    else:
      # if no broken letters, increment the count
      count += 1
      
  # Step 5: Return how many words we can type   
  return count

File: SOLUTIONS/LEETCODE/Vowels and Words/canBeTypedWords/canBeTypedWords.js
function canBeTypedWords(text, brokenLetters) {
  // Step 1: Put broken letters into a Set for quick lookup
  const brokenSet = new Set(brokenLetters)

  // Step 2: Split text into words (by space)
  const words = text.split(" ")

  // Step 3: Initialize counter
  let count = 0 

  // Step 4: Check each word
  for (let word of words) {
    // Assume the word is valid unless we find a broken letter
    let isValid = true 

    for (let char of words) {
      if (brokenSet.has(char)) {
        // Found a broken letter → word invalid
        isValid = false
        break; // stop checking this word
      }
    }

    if(isValid) count++
  }

  // Step 5: Return the result
  return count
}